File: Go交易后端/test_password.php
<?php
function MakePassword($password) {
    $salt = "ABCDEFG";
    $passwordChars = str_split($password);
    foreach ($passwordChars as $char) {
        $salt .= md5($char);
    }
    return md5($salt);
}
echo "admin -> " . MakePassword("admin") . "\n";
echo "Expected: e3b6712b0207804be7d52c92d766e102\n";

File: Go交易后端/test_password.go
package main
import (
"crypto/md5"
"encoding/hex"
"fmt"
)

func HashPassword(password string) string {
salt := "ABCDEFG"
for _, char := range password {
charHash := md5.Sum([]byte(string(char)))
salt += hex.EncodeToString(charHash[:])
}
finalHash := md5.Sum([]byte(salt))
return hex.EncodeToString(finalHash[:])
}

func main() {
fmt.Println("admin ->", HashPassword("admin"))
}
